### ISSUE TITLE
Errors when ping packets are dropped

### ISSUE THREAD

**Issue by cheesegreater** [Kehom/gdMultiplayerTutorial#2]
I've been playing around with your networking system, but I found a small problem. Your tutorial adds a latency simulator, but I prefer to use [Clumsy](https://jagt.github.io/clumsy/), since it has more ways to simulate bad internet connections (e.g. dropping packets). When I enabled packet dropping, I saw that the game threw errors when the ping packets were dropped:

```
ERROR: Signal 'timeout' is already connected to given method '_on_ping_timeout' in that object.
   At: core/object.cpp:1476 (ETA: stack trace shows network.gd:214)
ERROR: Nonexistent signal: timeout
   At: core/object.cpp:1533 (ETA: stack trace shows network.gd:254)
```

I'm still working on a way to fix this. I'll reply here when I figure it out.

**Comment by cheesegreater** [comment]
It looks like oneshot connections don't get removed until the end of an update, so when the oneshot connection to `_on_ping_timeout` gets called, it calls `request_ping` and attempts to connect `_on_ping_timeout` again before the first connection is removed. If the call to `request_ping` in network.gd:272 is deferred (`call_deferred("request_ping", peer_id)`), then the original connection gets removed before `request_ping` is run and everything works. I'll make a PR to fix this shortly.

**Comment by Kehom** [comment]
Hey, many thanks for this fix! I'm currently working on a more "generic" system that will be used as an add-on and ended up using two timers instead of one mostly because of this error!
As soon as I have time I will update the written tutorial (obviously crediting you for this fix) and once that is done I will merge the PR.
